src/leaderboard/views.py

```diff
@@ -74,3 +75,24 @@ def list(self, request, *args, **kwargs):
         if should_hide_scoreboard():
             return FormattedResponse({})
         return super(TeamListView, self).list(request, *args, **kwargs)
+
+
+class MatrixScoreboardView(APIView):
+    throttle_scope = 'leaderboard'
+
+    def get(self, request, *args, **kwargs):
+        if should_hide_scoreboard():
+            return FormattedResponse({})
+
+        scoreboard = []
+        teams = Team.objects.filter(is_visible=True).order_by('-leaderboard_points', 'last_score').prefetch_related(
+            Prefetch('solves', queryset=Solve.objects.filter(correct=True))
+        )
+        for team in teams:
+            scoreboard.append({
+                'id': team.id,
+                'name': team.name,
+                'points': team.leaderboard_points,
+                'solves': list(team.solves.values_list('challenge', flat=True))
+            })
```

Comment:
Is there any `aggregate` magicks we could apply here to avoid iterating all entries and creating a dict from them?

Reply:
I'd be surprised if there isn't